### PR TITLE
[Redesign] About page

### PR DESCRIPTION
**Pull request by scottbommarito**
This page is mostly the same with updated formatting and a new section at the top.

I changed the implementation of the script that puts heads into boxes to fix a bug, make it more readable, and use knockout.

image

@microsoftsam

### CODE REVIEW

File: src/Bootstrap/less/theme/page-about.less [hunk 1]
```diff
@@ -0,0 +1,25 @@
+.page-about {
+    h1 {
+        margin-bottom: 20px;
+    }
```
Comment (microsoftsam) [inline src/Bootstrap/less/theme/page-about.less]: I'm seeing this in a bunch of PRs. Why don't we make this the default when a h1/h2 shows up under a page .container div.

File: src/NuGetGallery/Views/Pages/About.cshtml [hunk 2]
```diff
@@ -1,102 +1,117 @@
...
-    <a href="https://github.com/NuGet/NuGetGallery/blob/master/LICENSE.txt">Apache 2 License</a>. This means
-    you can take the code, modify it and redistribute it freely, as long as you adhere to the terms of the license.
-</p>
-<p>
-    We also welcome contributions from the community. If you would like to contribute, check out our
-    <a href="https://github.com/NuGet/NuGetGallery/blob/master/CONTRIBUTING.md">Contributors Guide</a> on
-    our <a href="https://github.com/NuGet/NuGetGallery">GitHub Project Page</a>.
-</p>
+<div class="container main-container page-about">
+    <div class="row">
```
Comment (microsoftsam) [inline src/NuGetGallery/Views/Pages/About.cshtml]: <section

File: src/NuGetGallery/Views/Pages/About.cshtml [hunk 3]
```diff
@@ -1,102 +1,117 @@
...
-    The NuGet Gallery is <a href="http://opensource.org/">Open-Source Software</a> licensed under the
-    <a href="https://github.com/NuGet/NuGetGallery/blob/master/LICENSE.txt">Apache 2 License</a>. This means
-    you can take the code, modify it and redistribute it freely, as long as you adhere to the terms of the license.
-</p>
-<p>
-    We also welcome contributions from the community. If you would like to contribute, check out our
-    <a href="https://github.com/NuGet/NuGetGallery/blob/master/CONTRIBUTING.md">Contributors Guide</a> on
-    our <a href="https://github.com/NuGet/NuGetGallery">GitHub Project Page</a>.
-</p>
+<div class="container main-container page-about">
```
Comment (joelverhagen) [inline src/NuGetGallery/Views/Pages/About.cshtml]: `<section role="main" ...>`

**Comment by joelverhagen** [comment]
I think we should keep the **Special Thanks** section where it was (before the team). Their contributions are really awesome, we just get paid for this work ;).

Just my opinion.

File: src/NuGetGallery/Views/Pages/About.cshtml (hunk 2 above)
Comment (scottbommarito) [inline src/NuGetGallery/Views/Pages/About.cshtml]: Did you forget to type a message?

File: src/NuGetGallery/Views/Pages/About.cshtml (hunk 3 above)
Comment (scottbommarito) [inline src/NuGetGallery/Views/Pages/About.cshtml]: fixed

**Comment by scottbommarito** [comment]
@joelverhagen I would agree, but
image
takes up too much space to put anywhere but the bottom of the page.

File: src/NuGetGallery/Views/Pages/About.cshtml [hunk 4]
```diff
@@ -1,102 +1,140 @@
...
+                    var team = [];
+                    var contributors = [];
+
+                    // If contributors are in this map, they have been added.
+                    // Only display each contributor once.
+                    var contributionsByContributorMap = {};
+
+                    $.each(contributorsPerRepo, function (index, repoContributors) {
+                        $.each(repoContributors, function (index, repoContributor) {
+                            var isTeam = teamMap.hasOwnProperty(repoContributor.login);
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Pages/About.cshtml]: `isTeam` is unused if the `repoContributor` is already in the `contributionsByContributorMap`. Maybe move this down to right before the `if (isTeam) {` check?

File: src/NuGetGallery/Views/Pages/About.cshtml [hunk 5]
```diff
@@ -1,102 +1,140 @@
...
+            </p>
+
+            <h3>Special Thanks</h3>
+            <p>
+                A special thanks go out to some of our contributors who have made major contributions to the site.
+            </p>
+            <ul>
+                <li>
+                    <a href="https://github.com/NuGet/NuGetGallery/commits?author=robertmuehsig">Robert Muehsig</a> -
+                    Lots of bug fixes, up-for-grabs features and various assorted goodies
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Pages/About.cshtml]: Add a period at the end of this sentence.

File: src/NuGetGallery/Views/Pages/About.cshtml (hunk 4 above)
Comment (scottbommarito) [inline src/NuGetGallery/Views/Pages/About.cshtml]: moved into the `if`

File: src/NuGetGallery/Views/Pages/About.cshtml (hunk 5 above)
Comment (scottbommarito) [inline src/NuGetGallery/Views/Pages/About.cshtml]: fixed

File: src/NuGetGallery/Views/Pages/About.cshtml [hunk 6]
```diff
@@ -1,102 +1,140 @@
...
+            <h2>About the Gallery</h2>
+            <p>
+                The NuGet Gallery is <a href="http://opensource.org/">Open-Source Software</a> licensed under the
+                <a href="https://github.com/NuGet/NuGetGallery/blob/master/LICENSE.txt">Apache 2 License</a>. This means
+                you can take the code, modify it and redistribute it freely, as long as you adhere to the terms of the license.
+            </p>
+            <p>
+                We also welcome contributions from the community. If you would like to contribute, check out our
+                <a href="https://github.com/NuGet/NuGetGallery/blob/master/CONTRIBUTING.md">Contributors Guide</a> on
+                our <a href="https://github.com/NuGet/NuGetGallery">GitHub Project Page</a>.
```
Comment (microsoftsam) [inline src/NuGetGallery/Views/Pages/About.cshtml]: the "many many more" link needs more context for accessible users. 
One way to add context for screen readers would be possibly: 
```<a href="...">many many more<span class="sr-only"> user contributions</span></a>```

File: src/NuGetGallery/Views/Pages/About.cshtml [hunk 7]
```diff
@@ -1,102 +1,140 @@
...
+                <li>
+                    <a href="https://github.com/NuGet/NuGetGallery/commits?author=robertmuehsig">Robert Muehsig</a> -
+                    Lots of bug fixes, up-for-grabs features and various assorted goodies
+                </li>
+                <li>
+                    <a href="https://github.com/NuGet/NuGetGallery/commits?author=zippy1981">Justin Dearing</a> -
+                    Added the ability to send a copy of support emails to the originator.
+                </li>
+                <li>And <a href="https://github.com/NuGet/NuGetGallery/graphs/contributors">many many more!</a></li>
+            </ul>
```
Comment (microsoftsam) [inline src/NuGetGallery/Views/Pages/About.cshtml]: the "many many more" link needs more context for accessible users.
One way to add context for screen readers would be possibly:
```<a href="...">many many more<span class="sr-only"> user contributions</span></a>```

File: content/DEV/Team.json [hunk 8]
```diff
@@ -0,0 +1,64 @@
+﻿[
```
Comment (joelverhagen) [inline content/DEV/Team.json]: alphabetize? hard to see who's here and who's not

File: src/Bootstrap/less/theme/page-about.less [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    
+    h2, h3 {
+        margin-top: 40px;
+        margin-bottom: 20px;
+    }
+    
+    p {
```
Comment (joelverhagen) [inline src/Bootstrap/less/theme/page-about.less]: move to common style

File: src/Bootstrap/less/theme/page-about.less [hunk 10]
```diff
@@ -0,0 +1,25 @@
+.page-about {
+    h1 {
+        margin-bottom: 20px;
+    }
+    
+    h2, h3 {
+        margin-top: 40px;
```
Comment (joelverhagen) [inline src/Bootstrap/less/theme/page-about.less]: _try_ to move to common style, if it causes chaos don't worry about it

File: src/NuGetGallery/Views/Pages/About.cshtml [hunk 11]
```diff
@@ -1,102 +1,149 @@
...
+            <p>
+                And <a href="https://github.com/NuGet/NuGetGallery/graphs/contributors" aria-label="Read the full list of contributors on GitHub.">many many more!</a>
+            </p>
+            <div data-bind="template: { name: 'contributor-list-template', data: contributors }"></div>
+        </div>
+    </div>
+</section>
 
 @section BottomScripts {
     <script>
```
Comment (joelverhagen) [inline src/NuGetGallery/Views/Pages/About.cshtml]:
```
$(function() {
    ...
});`

File: src/NuGetGallery/Views/Pages/About.cshtml [hunk 12]
```diff
@@ -1,102 +1,149 @@
...
-        function addContributors($host, contributors, hrefBase) {
-            $.each(contributors, function (index, value) {
-                if (!value.avatar_url || !value.login) {
-                    return;
+        // This script downloads information about our team members and external contributors and displays them in an inline grid.
```
Comment (joelverhagen) [inline src/NuGetGallery/Views/Pages/About.cshtml]: put in external JS for now. Use DisplayPackage.cshtml and its JS as a model

File: src/Bootstrap/less/theme/page-about.less (hunk 10 above)
Comment (loic-sharma) [inline src/Bootstrap/less/theme/page-about.less]: I use `h3` on the report abuse page. If you do make this common, make sure it doesn't mess up that page.

File: content/DEV/Team.json (hunk 8 above)
Comment (scottbommarito) [inline content/DEV/Team.json]: done

File: src/Bootstrap/less/theme/page-about.less (hunk 10 above)
Comment (scottbommarito) [inline src/Bootstrap/less/theme/page-about.less]: I didn't see any `h3`, but I did see `h2`.

File: src/Bootstrap/less/theme/page-about.less (hunk 9 above)
Comment (scottbommarito) [inline src/Bootstrap/less/theme/page-about.less]: done

File: src/Bootstrap/less/theme/page-about.less (hunk 10 above)
Comment (scottbommarito) [inline src/Bootstrap/less/theme/page-about.less]: done

File: src/NuGetGallery/Views/Pages/About.cshtml (hunk 11 above)
Comment (scottbommarito) [inline src/NuGetGallery/Views/Pages/About.cshtml]: done

File: src/NuGetGallery/Views/Pages/About.cshtml (hunk 12 above)
Comment (scottbommarito) [inline src/NuGetGallery/Views/Pages/About.cshtml]: done

**Pull request merged**